### PR TITLE
Remove Figure/Axes/Axis deprecations from 3.3

### PR DESCRIPTION
**Pull request by QuLogic**
## PR Summary

Apparently, I missed a few of the `Axes`/`Axis` ones in #20331.

## PR Checklist

- [x] Has pytest style unit tests (and `pytest` passes).
- [x] Is [Flake 8](https://flake8.pycqa.org/en/latest/) compliant (run `flake8` on changed files to check).
- [n/a] New features are documented, with examples if plot related.
- [x] Documentation is sphinx and numpydoc compliant (the docs should [build](https://matplotlib.org/devel/documenting_mpl.html#building-the-docs) without error).
- [x] Conforms to Matplotlib style conventions (install `flake8-docstrings` and run `flake8 --docstring-convention=all`).
- [n/a] New features have an entry in `doc/users/next_whats_new/` (follow instructions in README.rst there).
- [x] API changes documented in `doc/api/next_api_changes/` (follow instructions in README.rst there).

### CODE REVIEW

File: lib/matplotlib/collections.py [hunk 1]
```diff
@@ -1634,32 +1634,17 @@ def switch_orientation(self):
         self._is_horizontal = not self.is_horizontal()
         self.stale = True
 
-    def set_orientation(self, orientation=None):
+    def set_orientation(self, orientation='horizontal'):
```
[the remark this answers is not in the document]
Comment (QuLogic) [reply lib/matplotlib/collections.py:1637]: Right, makes sense.

File: lib/matplotlib/contour.py [hunk 2]
```diff
@@ -720,8 +720,6 @@ class ContourSet(cm.ScalarMappable, ContourLabeler):
     %(contour_set_attributes)s
     """
 
-    ax = _api.deprecated("3.3")(property(lambda self: self.axes))
```
Comment (dstansby) [inline lib/matplotlib/contour.py:723]: I'm not sure there was a changelog entry for this removal?

**Review by dstansby — APPROVED** [review]

File: lib/matplotlib/contour.py (hunk 2 above)
Comment (QuLogic) [reply lib/matplotlib/contour.py:723]: Should be here: https://github.com/matplotlib/matplotlib/pull/20447/files#diff-891a63498fc22bcffb7555796d5a46cdda5e432360eaa4b13f459fc029af7858R52-R55

File: lib/matplotlib/gridspec.py [hunk 3]
```diff
@@ -625,19 +623,15 @@ def _from_subplot_args(figure, args):
         elif len(args) == 3:
             rows, cols, num = args
             if not (isinstance(rows, Integral) and isinstance(cols, Integral)):
-                _api.warn_deprecated("3.3", message=message)
-                rows, cols = map(int, [rows, cols])
+                raise ValueError(message)
```
[the remark this answers is not in the document]
Comment (QuLogic) [reply lib/matplotlib/gridspec.py]: Yep, done, and re-arranged a bit to simplify.

**Review by tacaswell — APPROVED** [review]

**Pull request merged**